Backpack/Rating/Classes/BPKRating.swift

```diff
@@ -0,0 +1,263 @@
+        size: BPKRatingSize = .default,
+        subtitle: String? = nil,
+        showScale: Bool = true,
+        titleView: UIView
+    ) {
+        self.value = value
+        self.ratingScale = ratingScale
+        self.size = size
+        self.showScale = showScale
+        self.titleView = titleView
+
+        super.init(frame: .zero)
+
+        self.subtitle = subtitle
+        self.accessibilityLabel = accessibilityLabel
+
+        setup()
+        updateLookAndFeel()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("NSCoding not supported")
+    }
+
+    private func setup() {
+        addSubview(horizontalStackView)
+        horizontalStackView.addArrangedSubview(ratingValueAndScaleStackView)
+        horizontalStackView.addArrangedSubview(titleSubtitleStackView)
+
+        ratingValueAndScaleStackView.addArrangedSubview(ratingValueLabel)
+        ratingValueAndScaleStackView.addArrangedSubview(ratingScaleLabel)
+
+        titleSubtitleStackView.addArrangedSubview(titleView)
+        titleSubtitleStackView.addArrangedSubview(subtitleLabel)
+
+        setupConstraints()
+    }
+
+    private func setupConstraints() {
+        horizontalStackView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            horizontalStackView.topAnchor.constraint(equalTo: topAnchor),
+            horizontalStackView.bottomAnchor.constraint(equalTo: bottomAnchor),
+            horizontalStackView.leadingAnchor.constraint(equalTo: leadingAnchor),
+            horizontalStackView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        ])
+    }
+
+    private func updateLookAndFeel() {
+        subtitleLabel.isHidden = subtitleLabel.text == nil
+        ratingScaleLabel.isHidden = !showScale
+        if showScale {
+            ratingScaleLabel.text = displayedScale(from: ratingScale)
+        }
+
+        updateSize()
+        ratingValueLabel.text = displayedValue(from: value, scale: ratingScale)
+        updateCustomTitleViewBottomConstraintIfNeeded()
+    }
+
+    private func updateSize() {
+        switch size {
+        case .default:
+            ratingValueLabel.fontStyle = .textLabel1
+            ratingScaleLabel.fontStyle = .textCaption
+            titleLabel.fontStyle = .textLabel1
+            subtitleLabel.fontStyle = .textCaption
+
+            titleSubtitleStackView.axis = .horizontal
+            titleSubtitleStackView.spacing = BPKSpacingMd
+            titleSubtitleStackView.alignment = .lastBaseline
+        case .large:
+            ratingValueLabel.fontStyle = .textHero5
+            ratingScaleLabel.fontStyle = .textBodyDefault
+            titleLabel.fontStyle = .textHeading5
+            subtitleLabel.fontStyle = .textBodyDefault
+
+            titleSubtitleStackView.axis = .vertical
+            titleSubtitleStackView.spacing = .zero
+            titleSubtitleStackView.alignment = .leading
+        }
+    }
```

Comment:
@frugoman I'm wondering if a styling struct (like a ColorSet) would be beneficial for this? Considering if we make different sizes/styles in the future.

Lines of code wise we won't be able to optimize I feel, but it might make customization easier in the future? 
(Would also move some of this code out of the component into the size enum file, which would follow similar patterns we have on other parts of the codebase.)